frontend/src/app/services/group.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import Group from '../models/group';
+import { SocketEvent } from '../utils/constants';
+import { SocketService } from './socket.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GroupService {
+  constructor(private http: HttpClient, private socketService: SocketService) {}
+
+  getById(groupID: string): Promise<Group> {
+    return this.http.get<any>('groups/' + groupID).toPromise();
```

Comment:
```suggestion
    return this.http.get<Group>('groups/' + groupID).toPromise();
```
Why not match the return types? Similar question for the other functions.